### PR TITLE
[FEATURE] Add github action to build and push release docker image

### LINKED ISSUE #2495: Outdated Docker image registry
**Describe the bug**
[Current documentation](https://docs.greatexpectations.io/en/latest/guides/how_to_guides/miscellaneous/how_to_use_official_docker_images.html) provides a link to the official registry, but the repository doesn't look like it's updated.

**To Reproduce**
1. Go to https://hub.docker.com/r/greatexpectations/great_expectations/tags
2. See only 0.12 versions

**Expected behavior**
See images for more recent great-expectation versions.

### PR DESCRIPTION
**Pull request by cedric-orange**
Changes proposed in this pull request:
- add github action to automate container image build

closes #2495

### Definition of Done
Please delete options that are not relevant.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

**Comment by cedric-orange** [comment]
@cla-bot check

**Comment by cdkini** [comment]
Hey @cedric-orange! Thank you so much for opening up this PR. Let me bring these proposed changes to our next triage meeting so I can determine next steps.

**Comment by cedric-orange** [comment]
Note that dockerfile should be upgraded also

File: .github/workflows/docker-image.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          
+      - name: Build and push
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ./docker/Dockerfile
+          build-args: PYTHON_DOCKER_TAG=3.7-buster
```
Comment (joshua-stauffer) [inline .github/workflows/docker-image.yml:41]: I suggest we use Python 3.9-buster as a base image

File: .github/workflows/docker-image.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        uses: docker/metadata-action@v3
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          
+      - name: Build and push
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ./docker/Dockerfile
```
Comment (joshua-stauffer) [inline .github/workflows/docker-image.yml:40]: add multi-arch build support here: `platforms: linux/amd64,linux/arm64`

File: .github/workflows/docker-image.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    runs-on: ubuntu-latest
+    
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+      
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+
```
Comment (joshua-stauffer) [inline .github/workflows/docker-image.yml:21]: To support multi-arch builds we'll need to add a QEMU step:
```yaml
- name: Set up QEMU
      id: qemu
      uses: docker/setup-qemu-action@v1
      with:
        image: tonistiigi/binfmt:latest
        platforms: all
```

File: .github/workflows/docker-image.yml (hunk 2 above)
Comment (joshua-stauffer) [reply .github/workflows/docker-image.yml:40]: these are the two i'm primarily familiar with, to support intel and apple silicon -- haven't tried to support windows natively, so unclear if there's a straightforward way to add that here.

File: .github/workflows/docker-image.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  build:
+    runs-on: ubuntu-latest
+    
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+      
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
```
Comment (joshua-stauffer) [inline .github/workflows/docker-image.yml:20]: add `install: true` to the buildx step to alias `docker build` to `docker buildx` -- in my experience, this was necessary to get my multi-arch builds to be successful.

**Review by joshua-stauffer — CHANGES_REQUESTED** [review]
Thanks for this contribution, @cedric-orange -- I left some thoughts, primarily around supporting multiple architectures.

**Comment by anthonyburdi** [comment]
Thank you so much for this contribution! We are planning to work on a strategy for our docker presence and this is a great start. We'll close this PR for now and @talagluck will add this work to our internal work tracking system and we may reopen this PR in the future as part of that effort.

**Pull request closed without merging**